scripts/setup-k8s-master.sh.in

```diff
@@ -55,41 +55,61 @@ kubeadm reset --force
 
 cat >/tmp/kubeadm.yaml <<EOF
 ---
-apiVersion: kubeadm.k8s.io/v1alpha2
-kind: MasterConfiguration
-kubernetesVersion: "$(cat /etc/kubernetes_community_ami_version)"
-api:
-  controlPlaneEndpoint: "${LB_DNS}:443"
-apiServerCertSANs:
-- "${LB_IPV4}"
-apiServerExtraArgs:
-  cloud-provider: "aws"
-nodeRegistration:
-  name: "${HOSTNAME}"
-  kubeletExtraArgs:
-    cloud-provider: "aws"
+apiVersion: kubeadm.k8s.io/v1beta1
 bootstrapTokens:
 - groups:
-  - "system:bootstrappers:kubeadm:default-node-token"
+  - system:bootstrappers:kubeadm:default-node-token
   token: "{{ClusterToken}}"
-  ttl: "0s"
+  ttl: 0s
   usages:
   - signing
   - authentication
-controllerManagerExtraArgs:
-  cloud-provider: "aws"
-  allocate-node-cidrs: "false"
+kind: InitConfiguration
+nodeRegistration:
+  criSocket: /var/run/dockershim.sock
+  name: "${HOSTNAME}"
+  kubeletExtraArgs:
+    cloud-provider: "aws"
+  taints:
+  - effect: NoSchedule
+    key: node-role.kubernetes.io/master
+---
+apiServer:
+  timeoutForControlPlane: 4m0s
+  certSANs:
+    - "${LB_IPV4}"
+  extraArgs:
+    cloud-provider: "aws"
+apiVersion: kubeadm.k8s.io/v1beta1
+certificatesDir: /etc/kubernetes/pki
+clusterName: kubernetes
+controlPlaneEndpoint: "${LB_DNS}:443"
+controllerManager:
+  extraArgs:
+    cloud-provider: "aws"
+    allocate-node-cidrs: "false"
+etcd:
+  local:
+    dataDir: /var/lib/etcd
+imageRepository: k8s.gcr.io
+kind: ClusterConfiguration
```

Comment:
similarly here, push both apiVersion and kind up to the top of the yaml doc to make it clearer to readers what k8s types we are using here.